src/library_dylink.js

```diff
@@ -1150,29 +1150,39 @@ var LibraryDylink = {
 #endif
   },
 
-  // Async version of dlopen.
-  _emscripten_dlopen_js__deps: ['$dlopenInternal', '$callUserCallback', '$dlSetError'],
-  _emscripten_dlopen_js: (handle, onsuccess, onerror, user_data) => {
-    /** @param {Object=} e */
-    function errorCallback(e) {
-      var filename = UTF8ToString(handle + {{{ C_STRUCTS.dso.name }}});
-      dlSetError(`'Could not load dynamic lib: ${filename}\n${e}`);
-      {{{ runtimeKeepalivePop() }}}
-      callUserCallback(() => {{{ makeDynCall('vpp', 'onerror') }}}(handle, user_data));
+  // Utility function to encapsulate promise handling with callbacks 
+function promiseToCallback(promise, successFnPtr, errorFnPtr, user_data) {
```

Comment:
You need to use the `$promiseToCallback: function(...) {` syntax here since this is inside a JS object.

You then need to add `$promiseToCallback` to the `__deps` for `_emscripten_dlopen_js`